Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/impl/apmlog/LogFilterControllerWithAPMLog.java

```diff
@@ -24,55 +24,193 @@
+            if (filterType == FilterType.EVENT_ATTRIBUTE_DURATION &&
+                    !logData.hasSufficientDurationVariant(mainAttribute, data)) {
+                Messagebox.show("Unable to filter on node duration as there's only one value.",
+                        "Filter error", Messagebox.OK, Messagebox.ERROR);
+                return;
+            }
+            parameters.put("key", mainAttribute);
+            parameters.put("value", data);
+        } else if (filterType == FilterType.ATTRIBUTE_ARC_DURATION) {
+            source = (String) param.get("source");
+            target = (String) param.get("target");
+            if (!logData.hasSufficientDurationVariant(mainAttribute, source, target)) {
+                Messagebox.show("Unable to filter on arc duration as there's only one value.",
+                        "Filter error", Messagebox.OK, Messagebox.ERROR);
+                return;
+            }
+            parameters.put("key", mainAttribute);
+            parameters.put("from", source);
+            parameters.put("to", target);
+        } else {
+            return;
+        }
+        Clients.showBusy("Launch Filter Dialog ...");
+
+        LogFilterRule rule = getLastMatchedRuleWithValues(filterType, parameters,
+                (List<LogFilterRule>) logData.getCurrentFilterCriteria());
+
+        EditorOption option = rule != null ? new EditorOption(filterType, rule) :
+                new EditorOption(filterType, parameters);
+
+        LogFilterRequest lfr = getRequestWithOption(option);
+        parent.getLogFilterPlugin().execute(lfr);
+        Clients.clearBusy();
+    }
+
+    private LogFilterRequest getRequestWithOption(EditorOption option) {
+        return new LogFilterRequest(this, parent.getSourceLogId(), parent.getTitle(),
+                logData.getOriginalAPMLog(), (List<LogFilterRule>) logData.getCurrentFilterCriteria(), option);
+    }
+
+    private LogFilterRequest getDefaultRequest() {
+        return new LogFilterRequest(this, parent.getSourceLogId(), parent.getTitle(),
+                logData.getOriginalAPMLog(), (List<LogFilterRule>) logData.getCurrentFilterCriteria());
+    }
+
+    private FilterType getFilterType(String payload) {
+        switch (payload) {
+            case "CaseTabID": return FilterType.CASE_ID;
+            case "CaseTabVariant": return FilterType.CASE_VARIANT;
+            case "EventTabAttribute": return FilterType.EVENT_EVENT_ATTRIBUTE;
+            case "CaseTabAttribute": return FilterType.CASE_EVENT_ATTRIBUTE;
+            case "CaseTabPerformance": return FilterType.DURATION;
+            case "CaseTabTimeframe": return FilterType.CASE_TIME;
+            case "CASE_SECTION_ATTRIBUTE_COMBINATION": return FilterType.CASE_SECTION_ATTRIBUTE_COMBINATION;
+            case "EVENT_ATTRIBUTE_DURATION": return FilterType.EVENT_ATTRIBUTE_DURATION;
+            case "ATTRIBUTE_ARC_DURATION": return FilterType.ATTRIBUTE_ARC_DURATION;
+        }
```

Comment:
Would it make sense to move the getFilterType method to be a method of FilterType?  Something like `FilterType parse(String s)`

Reply:
Perhaps, put the string representation as part of the enum.
https://github.com/apromore/ApromoreCore/blob/development/Apromore-Database/src/main/java/org/apromore/dao/model/HistoryEnum.java